XSLT/tsudctomods.xsl

```diff
@@ -0,0 +1,154 @@
+            
+            <!-- creator -->
+            <xsl:apply-templates select="dc:creator"/>
+            
+            <!-- originInfo> -->
+            <xsl:apply-templates select="dc:date"/>
+            
+            <!-- subject(s) -->
+            <xsl:apply-templates select="dc:subject"/>
+            
+            <!-- relatedItem -->
+            <xsl:apply-templates select="dc:source"/>
+            
+            <!-- publisher or recordContentSource -->
+            <recordInfo><recordContentSource>Tennessee State University</recordContentSource></recordInfo>
+            
+            <!-- rights or accessCondition -->
+            <accessCondition type="use and reproduction" xlink:href="http://rightsstatements.org/vocab/CNE/1.0/">Copyright Not Evaluated</accessCondition>
+            
+        </mods>
+    </xsl:template>
+    
+    <!-- title -->
+    <xsl:template match="dc:title">
+        <titleInfo><title><xsl:apply-templates/></title></titleInfo>
+    </xsl:template>
+    
+    <!-- identifier / object URL / typeOfResource based off collection -->
+    <xsl:template match='dc:identifier'>
+        <xsl:choose>
+            <xsl:when test="matches(., '[0-9]$')">
+                <location>
+                    <url usage="primary" access="object in context"><xsl:apply-templates/></url>
+                </location>
+            </xsl:when>
+            <xsl:when test="contains(., 'library-digital-collections/')">
+                <typeOfResource>still image</typeOfResource>
+            </xsl:when>
+            <xsl:when test="contains(., 'catalogues')">
+                <typeOfResource>text</typeOfResource>
+            </xsl:when>
+            <xsl:when test="contains(., 'yearbooks')">
+                <typeOfResource>text</typeOfResource>
+            </xsl:when>
+        </xsl:choose>
+    </xsl:template>
+    
+    <!-- abstract or thumbnail -->
+    <xsl:template match="dc:description">
+        <xsl:choose>
+            <xsl:when test="starts-with(., 'https://')">
+                <location>
+                    <url access="preview"><xsl:apply-templates/></url>
+                </location>
+            </xsl:when>
+            <xsl:otherwise>
+                <abstract><xsl:apply-templates/></abstract>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
+    
+    <!-- creator -->
+    <xsl:template match="dc:creator">
+        <name>
+            <namePart><xsl:apply-templates/></namePart>
+            <role>
+                <roleTerm authority="marcrelator" authorityURI="http://id.loc.gov/vocabulary/relators/cre">Creator</roleTerm>
+            </role>
+        </name>
+    </xsl:template>
+    
+    <!-- originInfo -->
+    <xsl:template match="dc:date">
+        <xsl:choose>
+            <xsl:when test="starts-with(., '19')">
+                <originInfo><dateCreated><xsl:apply-templates select="substring(., 1, string-length(.) -10)"/></dateCreated></originInfo>
```

Comment:
You can't do this against a Saxon 8.7 processor.  The substring node returns a value that is of type `xs:string`, but `xsl:apply-templates` expects a `node()`.

I should probably think about this, but I think this would work instead:

```
<xsl:copy-of select="substring(., 1, string-length(.) -10)"></xsl:copy-of>
```

Reply:
Thanks for this. I'm noticing that my when test is not necessary here so I've also removed it.